### PR TITLE
Add systemd support for SLES 12 +

### PR DESCRIPTION
**Pull request by zelaskov**
added support for systemd for SLES

tested on SLES 12.4 and puppet 5.5.22 and it works fine.
tested by running newrelic-infra agent with `ensure` and `ensure_service` arguments.

commented line 183 that caused error with installation of newrelic module --> `Error: Could not find resource 'Package[newrelic-infra]' for relationship on 'Service[newrelic-infra]' on node xyz`. after that module is correctly installed, so feedback from newrelic team would be needed.

### CODE REVIEW

File: manifests/agent.pp [hunk 1]
```diff
@@ -287,12 +287,20 @@
       provider => 'upstart',
     }
   } elsif $::operatingsystem == 'SLES' {
```
Comment (cristianciutea) [inline manifests/agent.pp:289]: won't this if branch apply for 11.4 also?

File: manifests/agent.pp [hunk 2]
```diff
@@ -287,12 +287,20 @@
...
       ensure => $service_ensure,
-      start  => '/etc/init.d/newrelic-infra start',
-      stop   => '/etc/init.d/newrelic-infra stop',
-      status => '/etc/init.d/newrelic-infra status',
+      start  => 'systemctl start newrelic-infra',
+      stop   => 'systemctl stop newrelic-infra',
+      status => 'systemctl status newrelic-infra',
+    }
+  } elsif $::operatingsystem == 'SLES' and $::operatingsystemmajrelease == '11.4' {
+    # Setup agent service for systemd service manager
```
Comment (cristianciutea) [inline manifests/agent.pp:298]: sysv-init

File: manifests/agent.pp [hunk 3]
```diff
@@ -180,7 +180,7 @@
               fail('New Relic Infrastructure agent is not yet supported on this platform')
             }
           }
-          Package['newrelic-infra'] -> Service['newrelic-infra']
+          #Package['newrelic-infra'] -> Service['newrelic-infra']
```
Comment (cristianciutea) [inline manifests/agent.pp:183]: is this required anymore?

**Comment by zelaskov** [comment]
hey @cristianciutea, thanks for reviewing a PR. I've made changes based on your comments. Could you review it again?

**Review by varas — APPROVED** [review]

**Pull request merged**